src/plugin/debugger/mhpdbgc.c

```diff
@@ -817,31 +817,38 @@ static int is_valid_program_name(const char *s)
     if (iscntrlDOS(*p))
       return 0;
   }
-  return 1;
+  return (s[0] != 0); // at least one character long
 }
 
-static const char *get_mcb_name2(uint16_t seg, uint16_t off)
+static const char *get_name_from_mcb(struct MCB *mcb)
 {
-  char *target = MK_FP32(seg, off);
   const char *dos = "DOS", *fre = "FREE";
   static char name[9];
+
+  if (mcb->owner_psp == 0)
+    return fre;
+  if (mcb->owner_psp == 8)
```

Comment:
Hmm, I've seen this 8 in freedos, but are
you sure its standard?

Reply:
Yep, tested okay in PC-DOS 3.00, MS-DOS 6.22, DR-DOS 6.00, FreeDOS 1.20 and FDPP